server/plugin.ts

```diff
@@ -47,8 +54,12 @@ export class SearchRelevancePlugin
 
     const router = core.http.createRouter();
 
-    const opensearchSearchRelevanceClient: ILegacyClusterClient =
-      core.opensearch.legacy.createClient('opensearch_search_relevance');
+    let opensearchSearchRelevanceClient: ILegacyClusterClient | undefined = undefined;
+    if (!dataSourceEnabled) {
+      opensearchSearchRelevanceClient = core.opensearch.legacy.createClient(
```

Comment:
I think we need to initialize the   opensearchSearchRelevanceClient even if data source is enabled, so that it will be uses as a fallback when data source id is empty string which indicates we are using local cluster